src/updateDisplay.js

```diff
@@ -0,0 +1,26 @@
+import pressedAnotherOperator from "./pressedAnotherOperator";
+
+const updateDisplay = (result, key, hasLastChar, lastEquation) => {
```

Comment:
Ah okay, yes. `hasLastChar`?

Reply:
That's a good catch! There's more though

Reply:
- 'hasLastChar' is also used by `pressedAnotherOperator', I moved it into the updateDisplay function, and the tests and app seem to work fine, but why? How does `pressedAnotherOperator' have access to 'hasLastChar'?
- I moved this line into `updateDisplay` so we wouldn't need to pass them in, but I would need to call them something else when they are used within the function because 'result' is a constant.
- I would also need to keep `const { result, lastEquation } = this.state;` within the onClick function in App.js because they are used elsewhere.

Reply:
Looks like you import and use `pressedAnotherOperator` in the `updateDisplay` file, so as long as `hasLastChar` can be found there, you should be good. The "one more" I was referring to wasn't for simplifying any parameters beyond `hasLastChar`. Any guesses at what other code could be moved into `updateDisplay`?

Reply:
Well we can't move the setState in there right, so hmm, are you hinting at the next block over? The `(button.key === "=")` if block?

Reply:
@grimxreaper I'm hinting at the return value

Reply:
You're saying the return value should be moved into `updateDisplay`...I'm confused 😅 do you mean this `var [tempResult, currentBtn]` inside of the onClick?

Reply:
Look carefully at what you return and how it's used. Is there a way you can return a single value rather than two values?

Reply:
@grimxreaper

Reply:
How about this, I combined `tempResult` and `key` by returning this from `updateDisplay`:  ` return result + key`)
and changing this in the `onClick`:

```
var currResult = updateDisplay(
          result,
          button.key,
          lastEquation
        );

        this.setState({
          result: currResult,
          lastEquation: "",
        });
```

Reply:
Yes!! Although is it really the currResult (don't be afraid to spell out current) or is it the nextResult since you just updated it?

Reply:
Oh true that! I just named it real fast and didn't think about that. Also I feel silly for not seeing this earlier. Will update now.

Reply:
All good. We are training your eyes to catch these things.